### ISSUE TITLE
yunyang先生你好，关于loss的问题我想请教下 

### ISSUE THREAD

**Issue by zhangyufei1995** [YunYang1994/tensorflow-yolov3#263]
第一，您这个loss_layer函数里面计算置信度损失（ conf_loss）的时候，tf.nn.sigmoid_cross_entropy_with_logits(labels=respond_bbox, logits=conv_raw_conf)

为何要用logits=conv_raw_conf（conv_raw_conf = conv[:, :, :, :, 4:5]） 而不是logits=pred_conf     = pred[:, :, :, :, 4:5]？ 这点不是很明白
第二  看了您前后的代码 改变挺大的  但是没明白这个GIOU解决的是什么问题？我看您输入的是预测的坐标和真实的坐标 返回的giou是啥呢 望回复 感激

**Comment by chentingkai** [comment]
@zhangyufei1995 问题1：为何要用logits=conv_raw_conf（conv_raw_conf = conv[:, :, :, :, 4:5]） 而不是logits=pred_conf = pred[:, :, :, :, 4:5]？这里因为在使用交叉熵的时候里面有一个sidmoid函数所以这里不再使用pred_conf 操作。 问题2：GIOU的操作应该是类似计算预测框和实际框的差别吧，两个框越吻合，GIOU损失越小

**Comment by zhouxianming** [comment]
> @zhangyufei1995 问题1：为何要用logits=conv_raw_conf（conv_raw_conf = conv[:, :, :, :, 4:5]） 而不是logits=pred_conf = pred[:, :, :, :, 4:5]？这里因为在使用交叉熵的时候里面有一个sidmoid函数所以这里不再使用pred_conf 操作。 问题2：GIOU的操作应该是类似计算预测框和实际框的差别吧，两个框越吻合，GIOU损失越小

@chentingkai 
你好请问在计算conf_loss时候：
![2019-11-26 22-11-43屏幕截图](https://user-images.githubusercontent.com/49632230/69640687-c470f800-1099-11ea-8ce2-c8b3178f1113.png)
![2019-11-26 22-12-24屏幕截图](https://user-images.githubusercontent.com/49632230/69640745-dbafe580-1099-11ea-88fe-7f039a9f970a.png)
按照focal loss 来计算损失的时候，当y=1,或者y=0的时候乘以前面的项不是都不一样，为什么代码中respond_bbox和respond_bgd前后景都乘以相同的conf_focal？
可否指点一下，谢谢。

**Comment by NoahFeng** [comment]
> > @zhangyufei1995 问题1：为何要用logits=conv_raw_conf（conv_raw_conf = conv[:, :, :, :, 4:5]） 而不是logits=pred_conf = pred[:, :, :, :, 4:5]？这里因为在使用交叉熵的时候里面有一个sidmoid函数所以这里不再使用pred_conf 操作。 问题2：GIOU的操作应该是类似计算预测框和实际框的差别吧，两个框越吻合，GIOU损失越小
> 
> @chentingkai
> 你好请问在计算conf_loss时候：
> ![2019-11-26 22-11-43屏幕截图](https://user-images.githubusercontent.com/49632230/69640687-c470f800-1099-11ea-8ce2-c8b3178f1113.png)
> ![2019-11-26 22-12-24屏幕截图](https://user-images.githubusercontent.com/49632230/69640745-dbafe580-1099-11ea-88fe-7f039a9f970a.png)
> 按照focal loss 来计算损失的时候，当y=1,或者y=0的时候乘以前面的项不是都不一样，为什么代码中respond_bbox和respond_bgd前后景都乘以相同的conf_focal？
> 可否指点一下，谢谢。

可以看看这个issue[https://github.com/wizyoung/YOLOv3_TensorFlow/issues/128](url)
